File: ios/FestivalConnection/Models/ChannelMessage.swift
import Foundation

struct ChannelMessage: Identifiable, Equatable {
    let id: String
    let channelId: String
    let senderPublicKeyHex: String
    let senderDisplayName: String
    let content: String
    let timestamp: Date

    init(id: String = UUID().uuidString, channelId: String, senderPublicKeyHex: String, senderDisplayName: String, content: String, timestamp: Date = Date()) {
        self.id = id
        self.channelId = channelId
        self.senderPublicKeyHex = senderPublicKeyHex
        self.senderDisplayName = senderDisplayName
        self.content = content
        self.timestamp = timestamp
    }

    @MainActor
    var isFromLocalUser: Bool {
        senderPublicKeyHex == IdentityManager.shared.publicKeyHex
    }
}

File: android/app/src/main/java/com/appxstudios/festivalconnection/models/ChannelMessage.kt
package com.appxstudios.festivalconnection.models

import java.util.UUID

data class ChannelMessage(
    val id: String = UUID.randomUUID().toString(),
    val channelId: String,
    val senderPublicKeyHex: String,
    val senderDisplayName: String,
    val content: String,
    val timestamp: Long = System.currentTimeMillis()
)
